snapshot-labs/snapshot
# feat: Support for Optimistic Quorum

Changes from https://github.com/snapshot-labs/snapshot/pull/4625 are also included in this PR,

so make sure it is merged to avoid confusion, with that PR you should be able to configure different types of quorum on proposals

This PR adds support for an optimistic quorum with which spaces can reject some proposal if the reach certain threshold 

### How to test:
- Post proposals with different with different quorum types
- default quorum should work same as before
- Optimistic quorum should only consider 2nd choice and reject proposal if it reaches `quorum` value
- for example if you go to [this proposal](https://snapshot-git-optimistic-quorum-snapshot.vercel.app/#/thanku.eth/proposal/0xcdca7ef4f76e5d5bf38aca6f0fccf8f0959a2119be236aefdce35d8c25d1fdb6)
![image](https://github.com/snapshot-labs/snapshot/assets/15967809/567da9ba-0754-440f-96c9-0d52eacbf7a8)
- on proposals pages you see the % of quorum rejection
![image](https://github.com/snapshot-labs/snapshot/assets/15967809/f4eee02f-0f4c-45bc-a26f-5e5b84992ac8)
